consumer.go

```diff
@@ -195,15 +199,36 @@ func (c *segmentConsumer) positionOf(m *kafka.Message) ([]byte, error) {
 	return p.json()
 }
 
-func (c *segmentConsumer) Ack() error {
-	// See issue related to this: https://github.com/ConduitIO/conduit-connector-kafka/issues/23
-	err := c.reader.CommitMessages(context.Background(), *c.lastMsgRead)
+func (c *segmentConsumer) Ack(position sdk.Position) error {
+	err := c.canAck(position)
+	if err != nil {
+		return fmt.Errorf("ack not possible: %w", err)
+	}
+	err = c.reader.CommitMessages(context.Background(), *c.unackMessages[0])
 	if err != nil {
 		return fmt.Errorf("couldn't commit messages: %w", err)
 	}
+	// remove the message from slice of unacknowledged messages
+	c.unackMessages = c.unackMessages[1:]
```

Comment:
... and we are reading and removing messages here (another goroutine). We need to set up a lock to make sure we don't run into race conditions.

Reply:
Good point with that. I've added that in [b38c192](https://github.com/ConduitIO/conduit-connector-kafka/pull/30/commits/b38c19232c7151199c21a737945802014ad3cd29).

Reply:
That should do the trick!